### ISSUE TITLE
Custom Inline Tags

### ISSUE THREAD

**Issue by rpowelll** [archriss/react-native-render-html#63]
As far as I can tell, even if a custom tag is inside something like a `<p>` tag, its rendered component won't be rendered inside that parent tag's `Text` node.

e.g. given this HTML markup

```html
<p>Foo <x-mytag>Bar</x-mytag> Baz</p>
```

I'd like to render something like this

```jsx
<Text>Foo <MyTag>Bar</MyTag> Baz</Text>
```

But instead what i get is this

```jsx
<Text>Foo</Text>
<MyTag>Bar</MyTag>
<Text>Baz</Text>
```

Is there any way to achieve this?

**Comment by Exilz** [comment]
Hi, I started working on this issue and I think I have a decent solution.

You'd be able to declare your custom tags like so :

```javascript
            renderers: {
                mytag: { renderer: myTagRenderer, wrapper: 'Text' }, // new way
                myothertag: myOtherTagRenderer // old regular way (still valid)
            }
```

If you supply a function, it behaves as it used to. If you use an object, your function must be in the `renderer` key so you can add other options.

I added the `wrapper` key, so you can set the wrapper you want for you custom component yourself.

**Please note that the `wrapper` is not a react-native component per-se, it's either `"View"` or `"Text"`. It's only used by the plugin to make sure views aren't nested inside texts to prevent crashes**.

The following renderers behave like this 

![](https://puu.sh/yFtyT/6af5a753c3.png)

Please tell me what you think about this solution, and I'll try to add it to the next version.

**Comment by rpowelll** [comment]
Sorry for the delayed response @Exilz, I've been off on vacation! I think this is probably the best solution in terms of API design. The only other solution I can think of is checking whether the returned node has an `inline` prop that's been set to `true`, but that feels a little too 'magical; to me.

**Comment by Exilz** [comment]
This feature landed in 3.6.0. Thanks for the constructive feedback.
